ray-operator/test/e2e/raycluster_test.go

```diff
@@ -121,3 +122,34 @@ func TestRayClusterSuspend(t *testing.T) {
 	g.Eventually(RayCluster(test, namespace.Name, rayCluster.Name), TestTimeoutMedium).
 		Should(WithTransform(StatusCondition(rayv1.RayClusterProvisioned), MatchCondition(metav1.ConditionTrue, rayv1.AllPodRunningAndReadyFirstTime)))
 }
+
+func TestRayClusterWithResourceQuota(t *testing.T) {
+	test := With(t)
+	g := NewWithT(t)
+
+	// Create a namespace
+	namespace := test.NewTestNamespace()
+
+	// Create a resource quota
+	CreateResourceQuota(test, namespace.Name, "test-quota", "0.1", "0.1Gi")
+
+	rayClusterAC := rayv1ac.RayCluster("raycluster-resource-quota", namespace.Name).WithSpec(newRayClusterSpec())
+
+	rayCluster, err := test.Client().Ray().RayV1().RayClusters(namespace.Name).Apply(test.Ctx(), rayClusterAC, TestApplyOptions)
+	g.Expect(err).NotTo(HaveOccurred())
+	LogWithTimestamp(test.T(), "Created RayCluster %s/%s successfully", rayCluster.Namespace, rayCluster.Name)
+
+	LogWithTimestamp(test.T(), "Waiting for RayCluster %s/%s to have ReplicaFailure condition", rayCluster.Namespace, rayCluster.Name)
+	g.Eventually(func() bool {
+		rc, err := RayCluster(test, namespace.Name, rayCluster.Name)()
+		if err != nil {
+			return false
+		}
+		for _, condition := range rc.Status.Conditions {
+			if condition.Type == "ReplicaFailure" && strings.Contains(condition.Message, "forbidden: exceeded quota") {
+				return true
+			}
+		}
+		return false
+	}, TestTimeoutShort).Should(BeTrue(), "Expected ReplicaFailure condition with message containing 'forbidden: exceeded quota'")
```

Comment:
Hi @han-steve,

Thank you for the new test!

```suggestion
	g.Eventually(RayCluster(test, namespace.Name, rayCluster.Name), TestTimeoutMedium).
		Should(WithTransform(StatusCondition(rayv1.RayClusterReplicaFailure), MatchCondition(metav1.ConditionTrue, ...)))
```

Could you also help us extend the existing `MatchCondition` or add a new matcher that can match condition message?

Reply:
Fixed!